tests/unit/handlers/test__monitored_resources.py

```diff
@@ -353,3 +343,40 @@ def test_detect_partial_data(self):
             # project id not returned from metadata serve
             # should be empty string
             self.assertEqual(resource.labels["project_id"], "")
+
+
+class Test_Add_Environmental_Labels(Test_Resource_Detection):
+    def setUp(self):
+        super().setUp()
+        self.record = logging.LogRecord("logname", None, None, None, "test", None, None)
+
+    def add_environment_labels_test_flow(self, resource_type, expected_labels):
+        resource = Resource(type=resource_type, labels={})
+        actual_labels = add_environment_labels(resource, self.record)
+        self.assertDictEqual(actual_labels, expected_labels)
+
+    def test_gae_label(self):
+        trace_id = "trace_id"
+        setattr(self.record, "_trace", trace_id)
+        self.add_environment_labels_test_flow(
+            _monitored_resources._GAE_RESOURCE_TYPE,
+            {_monitored_resources._GAE_TRACE_ID_LABEL: trace_id},
+        )
+
+    def test_cloud_run_job_label(self):
+        test_execution_id = "test_job_12345"
+        test_task_index = "1"
+        test_task_attempt = "12"
+
+        os.environ[_monitored_resources._CLOUD_RUN_EXECUTION_ID] = test_execution_id
+        os.environ[_monitored_resources._CLOUD_RUN_TASK_INDEX] = test_task_index
+        os.environ[_monitored_resources._CLOUD_RUN_TASK_ATTEMPT] = test_task_attempt
+
+        self.add_environment_labels_test_flow(
+            _monitored_resources._CLOUD_RUN_JOB_RESOURCE_TYPE,
+            {
+                _monitored_resources._CLOUD_RUN_JOBS_EXECUTION_NAME_LABEL: test_execution_id,
+                _monitored_resources._CLOUD_RUN_JOBS_TASK_INDEX_LABEL: test_task_index,
+                _monitored_resources._CLOUD_RUN_JOBS_TASK_ATTEMPT_LABEL: test_task_attempt,
+            },
+        )
```

Comment:
maybe also a test for other resource types?

Suggestion: This seems like it could be a good place for `pytest.mark.paramaterize` to really scale up different test cases:

```
@pytest.mark.paramaterize("env,record_fields,resource,expected_labels", [
    {}, {"_trace":"trace"}, "gae_app", {"appengine.googleapis.com/trace_id": "trace"},
   {"CLOUD_RUN_EXECUTION": "test_name"}, {}, "cloud_run_job", {"run.googleapis.com/execution_name": "test_name"}
  ...
])
def test_add_environment_labels(env, resource, expected_labels):
  ...
```